api/wasm.go

```diff
@@ -355,6 +355,24 @@ type Function interface {
 	// WithCloseOnContextDone on wazero.RuntimeConfig for detail. See examples in context_done_example_test.go for
 	// the end-to-end demonstrations of how these terminations can be performed.
 	Call(ctx context.Context, params ...uint64) ([]uint64, error)
+
+	// CallWithStack is an optimized variation of Call that saves memory
+	// allocations when the stack slice is large enough to hold the maximum
+	// length of function parameters and results.
+	//
+	// The caller adds parameters in order to the stack, and reads any results
+	// in order from the result stack, except in the error case.
+	//
+	// # Notes
+	//
+	//   - When the stack parameter is large enough to hold the max count of
+	//     parameters and results, it is returned. Otherwise, a new slice is
+	//     allocated and returned. This means the caller should always prefer
+	//     using the result slice for subsequent calls.
+	//   - This is similar to GoFunc, except for using calling functions
```

Comment:
Let's use GoModuleFunction as it has more docs on it.

```suggestion
	//   - This is similar to GoModuleFunction, except for using calling functions
```